qiskit_ibm_runtime/utils/noise_learner_result.py

```diff
@@ -14,47 +14,136 @@
+    :math:`E(ρ) = e^{\sum_j r_j D_{P_j}}(ρ)` generated by Pauli Lindblad dissipators
+    :math:`D_P(ρ) = P ρ P - ρ`, where :math:`P_j` are N-qubit :class:`~.Pauli` operators.
+
+    The list of Pauli generator terms are stored as a :class:`~.PauliList` and can be
+    accessed via the :attr:`generators` attribute. The array of dissipator rates
+    :math:`r_j` can be accessed via the :attr:`rates` attribute.
+
+    The equivalent Pauli error channel can be constructed as a composition
+    of single-Pauli channel terms
+
+    .. math::
+
+        E = e^{\sum_j r_j D_{P_j}} = \prod_j e^{r_j D_{P_j}}
+        = prod_j \left( (1 - p_j) S_I + p_j S_{P_j} \right)
+
+    where :math:`p_j = \frac12 - \frac12 e^{-2 r_j}`.
 
     Args:
-        circuit: A circuit whose noise has been learnt.
-        qubits: The labels of the qubits in the given circuit.
-        generators: The list of generators for the Pauli errors.
-        rates: The rates for the given Pauli error generators.
+        generators: A list of the Pauli Lindblad generators for the error channel.
+        rates: A list of the rates for the Pauli-Lindblad ``generators``.
 
     Raises:
-        ValueError: If ``circuit``, ``qubits``, and ``generators`` have mismatching number of
-            qubits.
         ValueError: If ``generators`` and ``rates`` have different lengths.
     """
 
-    circuit: QuantumCircuit
-    qubits: Sequence[int]
-    generators: PauliList
-    rates: Sequence[float]
+    def __init__(self, generators: PauliList, rates: Sequence[float]) -> None:
+        self._generators = generators
+        self._rates = np.asarray(rates, dtype=float)
 
-    def __post_init__(self) -> None:
-        if len({self.circuit.num_qubits, len(self.qubits), self.generators.num_qubits}) != 1:
-            raise ValueError("Mistmatching numbers of qubits.")
-        if len(self.generators) != len(self.rates):
-            msg = f"``generators`` has length {len(self.generators)}, but "
-            msg += f"``rates`` has length {len(self.rates)}."
+        if len(generators) != len(self._rates):
+            msg = f"``generators`` has length {len(generators)} "
+            msg += f"but ``rates`` has length {len(self.rates)}."
             raise ValueError(msg)
 
+    @property
+    def generators(self) -> PauliList:
+        r"""
+        The Pauli Lindblad generators of this :class:`~.PauliLindbladError`.
+        """
+        return self._generators
+
+    @property
+    def rates(self) -> NDArray[np.float64]:
+        r"""
+        The Lindblad generator rates of this quantum error.
+        """
+        return self._rates
+
+    @property
+    def num_qubits(self) -> int:
+        r"""
+        The number of qubits in this :class:`~.PauliLindbladError`.
+        """
+        return self.generators.num_qubits
+
+    def __repr__(self) -> str:
+        return f"PauliLindbladError(generators={self.generators}, rates={self.rates.tolist()})"
+
+
+class LayerError:
```

Comment:
What's the argument for for having both `circuit` and `qubits`? Why can't the circuit just be ISA, and only refer to physical qubits?

The circuit inputs to the primitives are ISA, which makes me prefer to reference everything without a layout. qubit mappings tend to be confusing.

Reply:
I don't have a good answer for that, rather than "this is how things work in pec-runtime."
Elaborating on this, we'd like to have "block" here, but we don't have a well-established concept of block in qiskit-runtime yet. So next in line is "circuit" and "qubits," which is what a block is made of (where I am assuming that the definition of block that we have in pec-runtime is here to stay). Having ISA circuits with physical qubits would not be in line with the definition of block in pec-runtime, and I worry that we'll end up having too many concepts of block here and there

Also interested in hearing @chriseclectic's opinion though

Reply:
This is non-blocking for this PR, I just wanted to poke at it. I understand a bit better now that qubits+circuit is more or less supposed to mimic a `CircuitInstruction`, if we had available to us proper circuit blocks.

Reply:
The qubits referrer to noise / twirled qubits of a block, which could be in parallel with other blocks, so is very far from an ISA circuit block which would always be a full width layer. Eg if you twirled a two qubit gate on qubits (a,b) out of N this would be a 2-qubit circuit with qubits (a, b).If you put in ISA circuits you are back to the issue of not knowing which subset of qubits was twirled and hence which ones the noise model applies to, unless you pad all your generators to be ISA generators too, but then you can'tdo things like noise model stitching with independent parallel blocks etc